File: wallet/src/main/java/com/binance4j/wallet/dto/DustLog.java
package com.binance4j.wallet.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Logs of assets converted in BNB.
 * 
 * @param total              Total counts of exchange.
 * @param userAssetDribblets Dribblets.
 */
@ApiModel("Logs of assets converted in BNB.")
public record DustLog(@ApiModelProperty("Total counts of exchange.") int total,
		@ApiModelProperty("Dribblets.") List<UserAssetDribblets> userAssetDribblets) {
}

File: connectors/src/main/kotlin/com/binance4j/wallet/dto/DustLog.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Logs of assets converted in BNB.
 *
 * @property total Total counts of exchange.
 * @property userAssetDribblets Dribblets.
 */
@ApiModel("Logs of assets converted in BNB.")
data class DustLog(
	@ApiModelProperty("Total counts of exchange.") val total: Int
	@ApiModelProperty("Dribblets.")
@JsonProperty
var userAssetDribblets: List<UserAssetDribblets> = emptyList()
)
